src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -380,4 +380,16 @@
   <data name="DeserializePolymorphicInterface" xml:space="preserve">
     <value>Deserialization of interface types is not supported. Type '{0}'</value>
   </data>
+  <data name="FormatByte" xml:space="preserve">
+    <value>The JSON value is either of incorrect numeric format, or too large or too small for a Byte.</value>
+  </data>
+  <data name="FormatInt16" xml:space="preserve">
+    <value>The JSON value is either of incorrect numeric format, or too large or too small for an Int16.</value>
+  </data>
+  <data name="FormatSByte" xml:space="preserve">
+    <value>The JSON value is either of incorrect numeric format, or too large or too small for an SByte.</value>
+  </data>
+  <data name="FormatUInt16" xml:space="preserve">
+    <value>The JSON value is either of incorrect numeric format, or too large or too small for a UInt16.</value>
+  </data>
```

Comment:
Nit: grammatically, these should be reworded slightly, e.g.
```
The JSON value either uses an incorrect numeric format or is too large or too small for a Byte.
```

Reply:
Also, would it make sense to include in the error message the value that couldn't be parsed?

Reply:
> Nit: grammatically, these should be reworded slightly, e.g.

> Also, would it make sense to include in the error message the value that couldn't be parsed?

@stephentoub does this sound ok? Will make changes in all the places accordingly.

```
The JSON value could not be parsed as a number, uses incorrect numeric format or is too large or too small for a Byte.
```

Reply:
Functionally speaking, "uses an incorrect numeric format" is redundant with "could not be parsed as a number".

`sbyte.Parse`'s failures are "Input string was not in a correct format." or "Value was either too large or too small for a signed byte."

Putting them together into one, I'd go with

```
The JSON value either was not in a supported format, or is out of bounds for a signed byte.
```
("out of bounds" to remove the second "or")

It looks like the primitive Parse methods use the type name (e.g. Int32) except for byte/sbyte, where it's "unsigned byte" (lowercase) and "signed byte" (lowercase).  https://github.com/dotnet/coreclr/blob/29810a78e5b93d8da9fb921d096226d249fc75a5/src/System.Private.CoreLib/Resources/Strings.resx#L3130-L3162

Reply:
> The JSON value either was not in a supported format, or is out of bounds for a signed byte.

Should the tense be same in both clauses? We have `was` and `is` here. Intrigued, I searched and came across this,
https://grammar.collinsdictionary.com/easy-learning/joining-clauses

```
<quote>
When it is used in this way either must come in one of these places:
- before the subject in the first clause of the group.
- in front of the main verb and after any auxiliary verb.
</quote>
```
Looks like `either` needs to be before the `subject` in this case? E.g.,
```
Either the JSON value is not in a supported format, or is out of bounds for a signed byte.
```

Reply:
Yeah, is/is seems better :)

Reply:
```
Either the JSON value is not in a supported format, or is out of bounds for a signed byte.
```

That exception message looks good to me.

Reply:
I suggest to include the invalid value. Can be pretty helpful for offline analysis. Quite often for example in random unit test failures in corefx we end up going in to add more logging of the actual value. Up to you

Reply:
> I suggest to include the invalid value.

Do you mean the actual JSON string? I think transcoding the ValueSpan/ValueSequence back to UTF-16 string and appending that might be useful.

```C#
string json = "[12345, 1]";
byte[] utf8 = Encoding.UTF8.GetBytes(json);
var reader = new Utf8JsonReader(utf8, ...);
reader.Read(); // Start Array
reader.Read(); // Number

// This method call will throw with the message:
// "Either the JSON value is not in a supported format, or is out of bounds for a signed byte."
// What should we append here? Re-parse "12345" as a larger numeric type and append the value to the exception message?
// OR, take the raw utf-8 bytes that represent 12345, transcode it back to UTF-16 string, and append that to the exception message?
reader.GetByte();

```

Reply:
@danmosemsft @ahsonkhan given the 6/26 deadline and that we haven't been including the actual JSON fragement into the error message until now, I am inclined to change the error message to the `Either the JSON value is...` string discussed above. For more changes perhaps we can have another issue?

Reply:
That sounds good to me.